### PR TITLE
General: Add allowed_blocks field to block registration and REST API

### PR DESCRIPTION
**Pull request by jsnajdr**
There is a new block.json field called `allowedBlocks`, added in Gutenberg in https://github.com/WordPress/gutenberg/pull/58262. This is a companion patch to support this new field also on the server. It ensures that the following works:

1.  If the `allowedBlocks` field is defined in block's `block.json`, the objects registered in `WP_Block_Type_Registry` will have the `allowed_blocks` field.
2. The `allowed_blocks` field is returned by the `/wp/v2/block-types` REST endpoint.
3. The `allowedBlocks` field will be included in the boostrapped block registration data in the `unstable__bootstrapServerSideBlockDefinitions` inline script. 

Trac ticket: [https://core.trac.wordpress.org/ticket/60403](https://core.trac.wordpress.org/ticket/60403)

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by gziolo** [comment]
`allowed_blocks` should be also explicitly defined in `WP_Block_Type` class.

**Comment by jsnajdr** [comment]
> `allowed_blocks` should be also explicitly defined in `WP_Block_Type` class.

Done ✅

**Comment by gziolo** [comment]
CI is unhappy about something. Other than that, I have enough to commit it tomorrow.

File: src/wp-includes/class-wp-block-type.php [hunk 1]
```diff
@@ -68,6 +68,14 @@ class WP_Block_Type {
 	 */
 	public $ancestor = null;
 
+	/**
+	 * Limits which block types can be inserted as children of this block type.
+	 *
+	 * @since 6.5.0
+	 * @var string[]|null
+	 */
+	public allowed_blocks = null;
```
Comment (gziolo) [inline src/wp-includes/class-wp-block-type.php]:
```suggestion
	public $allowed_blocks = null;
```

Comment (jsnajdr) [reply src/wp-includes/class-wp-block-type.php]: Yes that was it 🙂 I'm bad at PHP.

**Comment by gziolo** [comment]
Committed with https://core.trac.wordpress.org/changeset/57521.

**Pull request closed without merging**